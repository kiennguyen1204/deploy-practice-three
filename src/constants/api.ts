export const URL_API = {
  BASE_URL: process.env.VITE_BASE_URL,
  PRODUCTS_URL: '/products'
};

const PRODUCTS_API = `${URL_API.BASE_URL}${URL_API.PRODUCTS_URL}`;

export { PRODUCTS_API };
