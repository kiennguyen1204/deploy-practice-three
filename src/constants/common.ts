export const MAX_STAR = 5;
