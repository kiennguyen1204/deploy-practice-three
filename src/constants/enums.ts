export enum BUTTON_VARIANT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary'
}

export enum BUTTON_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
  EXTRA_LARGE = 'xl'
}

export enum BUTTON_COLOR {
  SUCCESS = 'success',
  WARNING = 'warning',
  DEFAULT = 'default'
}

export enum BUTTON_COLOR_TEXT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary'
}

export enum BUTTON_TYPE {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset'
}

export enum INPUT_VARIANT {
  DEFAULT = 'default',
  PRIMARY = 'primary'
}

export enum INPUT_TYPE {
  TEXT = 'text',
  NUMBER = 'number'
}

export enum INPUT_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
  EXTRA_LARGE = 'xl'
}

export enum STATUS {
  SUCCESS = 'success',
  ERROR = 'error'
}

export enum TYPE_LINKS {
  QUICK = 'quick',
  MORE = 'more',
  PAYMENT = 'payment'
}

export enum OPTION_WEED {
  FLOWERS,
  MUSHROOMS,
  CONCENTRACE
}

export enum STEPS_CART {
  SHOPPING_CART,
  CHECKOUT,
  ORDER_COMPLETE
}

export enum SORT_TYPE {
  ASC = 'price_asc',
  DESC = 'price_desc'
}