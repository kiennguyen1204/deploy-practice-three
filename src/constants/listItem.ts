export const LIST_ITEM_NAVIGATION = [
  { name: 'Shop All', to: '/about' },
  {
    name: 'Flower',
    to: ''
  },
  { name: 'Edibles', to: '' },
  {
    name: 'Concentrates',
    to: ''
  },
  { name: 'Mushrooms', to: '' },
  {
    name: 'Promotions/Bundles',
    to: ''
  },
  {
    name: 'Support',
    to: ''
  },
  { name: 'Rewards', to: '' },
  { name: 'Blog', to: '' }
];

export const DATA_INFO = [
  {
    title: 'Indica',
    image: '/src/assets/images/indica.png',
    icon: '/src/assets/images/sleeping_face.svg',
    description: `This type of cannabis is commonly taken by those who want to sink into a state of total relaxation in every limb. This reduces stress overall and can take your worries and fatigue away. Because of its relaxing effects, it is suggested to use this type of cannabis at night. It is particularly recommended for people who have trouble sleeping, be it due to insomnia, pain or other associated sleep issues.`,
    to: '/indica',
    shop: 'Shop Indica'
  },
  {
    title: 'Sativa',
    image: '/src/assets/images/indica.png',
    icon: '/src/assets/images/sun.svg',
    description:
      'Contrary to the deep all-body relaxation that comes with the use of indica strains, sativas are known for increasing the feeling of creativity, enhancing focus and lessening anxiety. This is more of a mind-centered high in terms of how and where you will feel the effects. Given its stimulating nature, it is recommended to use this during the day.',
    to: '/sativa',
    shop: 'Shop Sativa'
  },
  {
    title: 'Hybrids',
    image: '/src/assets/images/hybrids.png',
    description: `Hybrid strains are just that – they combine the head-focused high effects of sativas with the bodily relaxation of indicas. This is ideal for people who really want to sooth any fatigue and worries while also clearing the mind ready to focus fresh on something new. Hybrids are expertly tailored to result in strains with specific effects. Useful strain vocabulary to familiarize yourself with when it comes to hybrids includes sativa-dominant (full, bodily relaxation), indica-dominant (boosting creativity, increasing mood and lessening anxiety) and balanced (the best of both worlds).`,
    top: '/hybrids',
    shop: 'Shop Hybrids'
  }
];

export const LIST_ITEM_EDUCATION = [
  {
    image: '/src/assets/images/education/info.webp',
    title: '12 Mistakes To Avoid When Buying Cannabis Online',
    subtitle: 'January 24, 2023',
    description:
      'Buying cannabis online can be a great way to get your hands on the products you need without leaving the comfort of your home. But …',
    to: '/home'
  },
  {
    image: '/src/assets/images/education/weight.webp',
    title: 'How To Store Cannabis and Keep it Fresh and Potent?',
    subtitle: 'January 20, 2023',
    description:
      'Cannabis packaging has advanced dramatically in recent years, whether your state has a medicinal marijuana programme, legal adult-use weed, or both. Most ...',
    to: '/home'
  },
  {
    image: '/src/assets/images/education/weight.webp',
    title: 'The Ultimate Guide to Checking the Quality of Cannabis – 10 Industry Leading Tips',
    subtitle: 'January 19, 2023',
    description:
      'Quality cannabis is a term used to describe cannabis products that meet specific standards of excellence. It is essential to understand what quality cannabis means, …',
    to: '/home'
  }
];
