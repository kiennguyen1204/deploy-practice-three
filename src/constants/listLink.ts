import Mastercard from 'assets/images/master_card.svg';
import Visa from 'assets/images/visa.svg';
import Bitcoin from 'assets/images/bitcoin.svg';
import Interac from 'assets/images/interac.svg';

export const QUICK_LINKS = [
  {
    name: 'Track Your Order',
    to: '/home'
  },
  {
    name: 'Shop All',
    to: '/home'
  },
  {
    name: 'Flower',
    to: '/home'
  },
  {
    name: 'Edibles',
    to: '/home'
  },
  {
    name: 'Concentrates',
    to: '/home'
  },
  {
    name: 'Refunds',
    to: '/home'
  },
  {
    name: 'Mushrooms',
    to: '/home'
  },
  {
    name: 'Promotions / Bundles',
    to: '/home'
  },
  {
    name: 'Promotions / Bundles',
    to: '/home'
  },
  {
    name: 'Support',
    to: '/home'
  },
  {
    name: 'Reward',
    to: '/home'
  },
  {
    name: 'Blog',
    to: '/home'
  },
  {
    name: 'Shipping Faq',
    to: '/home'
  }
];

export const MORE_LINKS = [
  {
    name: 'Buy weed online in Canada',
    to: '/'
  },
  {
    name: 'Buy weed online in New Brunswick',
    to: '/'
  },
  {
    name: 'Buy weed online in Prince Edward Island',
    to: '/'
  },
  {
    name: 'Buy weed online in Northwest Territories',
    to: '/'
  },
  {
    name: 'Buy weed online in Saskatchewan',
    to: '/'
  },
  {
    name: 'Buy weed online in Manitoba',
    to: '/'
  },
  {
    name: 'Buy weed online in Quitebec',
    to: '/'
  },
  {
    name: 'Buy weed online in British Columbia',
    to: '/'
  },
  {
    name: 'Buy weed online in Ontario',
    to: '/'
  },
  {
    name: 'Buy weed online in Alberta',
    to: '/'
  }
];

export const PAYMENT_METHODS = [
  {
    name: 'mastercard',
    image: Mastercard
  },
  {
    name: 'visa',
    image: Visa
  },
  {
    name: 'bitcoin',
    image: Bitcoin
  },
  {
    name: 'interac',
    image: Interac
  }
];
