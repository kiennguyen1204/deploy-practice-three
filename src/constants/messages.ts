export const MESSAGES = {
  EMPTY: 'No suitable data found!',
  EMPTY_FILED: 'Please fill in this field',
  FAIL_TO_FETCH: 'Something went wrong!! Please reload page.',
  ADD_TO_CART_SUCCESS: 'Add to cart successful!',
  ADD_TO_CART_FAIL: 'Add to cart fail!',
  REMOVE_FROM_CART_SUCCESS: 'Remove from cart successful!',
  REMOVE_FROM_CART_FAIL: 'Remove from cart fail!',
  CHANGE_QUANTITY_ERROR: 'Quantity cannot be changed',
  ADD_NEW_PRODUCT_SUCCESS: 'Add product success',
  ADD_NEW_PRODUCT_FAIL: 'Add product success',
  UPDATE_PRODUCT_SUCCESS: 'Add product success',
  UPDATE_PRODUCT_FAIL: 'Update product fail',
  REMOVE_PRODUCT_SUCCESS: 'Remove product success',
  REMOVE_PRODUCT_FAIL: 'Remove product fail',
  EXIST_PRODUCT: 'Product already exist'
};
