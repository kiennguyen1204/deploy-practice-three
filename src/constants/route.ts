const ROUTE = {
  HOMEPAGE: '/',
  PRODUCT: '/products/:productId',
  CATEGORY_PAGE: '/category',
  CART_PAGE: '/cart'
};

export default ROUTE;
