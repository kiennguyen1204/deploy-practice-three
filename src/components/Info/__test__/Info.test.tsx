import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

// constants
import { DATA_INFO } from 'constants/listItem';

// components
import InfoBottom from 'components/Info';

describe('InfoBottom', () => {
  beforeEach(() => {
    render(
      <MemoryRouter>
        <InfoBottom />
      </MemoryRouter>
    );
  });

  it('renders the correct number of BoxInfoItem components', () => {
    const boxInfoItems = screen.getAllByTestId('box-info-item');

    expect(boxInfoItems).toHaveLength(DATA_INFO.length);
  });

  it('passes the correct props to BoxInfoItem components', () => {
    DATA_INFO.forEach((data, index) => {
      const boxInfoItem = screen.getAllByTestId('box-info-item')[index];

      expect(boxInfoItem).toBeInTheDocument();

      // Check the title prop
      expect(boxInfoItem).toHaveTextContent(data.title);
    });
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <InfoBottom />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
