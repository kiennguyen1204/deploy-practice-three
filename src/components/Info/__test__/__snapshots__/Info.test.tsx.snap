// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBottom renders correctly 1`] = `
<DocumentFragment>
  <section
    class="box-info-bottom-container"
  >
    <div
      class="box-container"
    >
      <section
        class="item-box"
        data-testid="box-info-item"
      >
        <picture
          class="flex-row-container card-image"
        >
          <img
            alt="Indica"
            src="/src/assets/images/indica.png"
          />
          <img
            alt="Indica Icon"
            aria-hidden="true"
            class="flex-container icon-image"
            src="/src/assets/images/sleeping_face.svg"
          />
        </picture>
        <h2
          class="title"
          data-testid="title"
        >
          Indica
        </h2>
        <p
          class="description"
          data-testid="description"
        >
          This type of cannabis is commonly taken by those who want to sink into a state of total relaxation in every limb. This reduces stress overall and can take your worries and fatigue away. Because of its relaxing effects, it is suggested to use this type of cannabis at night. It is particularly recommended for people who have trouble sleeping, be it due to insomnia, pain or other associated sleep issues.
        </p>
        <a
          aria-label="Visit Shop Indica"
          class="shop"
          data-testid="shop"
          href="/indica"
        >
          Shop Indica
        </a>
      </section>
      <section
        class="item-box"
        data-testid="box-info-item"
      >
        <picture
          class="flex-row-container card-image"
        >
          <img
            alt="Sativa"
            src="/src/assets/images/indica.png"
          />
          <img
            alt="Sativa Icon"
            aria-hidden="true"
            class="flex-container icon-image"
            src="/src/assets/images/sun.svg"
          />
        </picture>
        <h2
          class="title"
          data-testid="title"
        >
          Sativa
        </h2>
        <p
          class="description"
          data-testid="description"
        >
          Contrary to the deep all-body relaxation that comes with the use of indica strains, sativas are known for increasing the feeling of creativity, enhancing focus and lessening anxiety. This is more of a mind-centered high in terms of how and where you will feel the effects. Given its stimulating nature, it is recommended to use this during the day.
        </p>
        <a
          aria-label="Visit Shop Sativa"
          class="shop"
          data-testid="shop"
          href="/sativa"
        >
          Shop Sativa
        </a>
      </section>
      <section
        class="item-box"
        data-testid="box-info-item"
      >
        <picture
          class="flex-row-container card-image"
        >
          <img
            alt="Hybrids"
            src="/src/assets/images/hybrids.png"
          />
          <img
            alt="Hybrids Icon"
            aria-hidden="true"
            class="flex-container icon-image"
            src="/src/assets/images/default_image.webp"
          />
        </picture>
        <h2
          class="title"
          data-testid="title"
        >
          Hybrids
        </h2>
        <p
          class="description"
          data-testid="description"
        >
          Hybrid strains are just that â€“ they combine the head-focused high effects of sativas with the bodily relaxation of indicas. This is ideal for people who really want to sooth any fatigue and worries while also clearing the mind ready to focus fresh on something new. Hybrids are expertly tailored to result in strains with specific effects. Useful strain vocabulary to familiarize yourself with when it comes to hybrids includes sativa-dominant (full, bodily relaxation), indica-dominant (boosting creativity, increasing mood and lessening anxiety) and balanced (the best of both worlds).
        </p>
        <a
          aria-current="page"
          aria-label="Visit Shop Hybrids"
          class="shop active"
          data-testid="shop"
          href="/"
        >
          Shop Hybrids
        </a>
      </section>
    </div>
  </section>
</DocumentFragment>
`;
